### PR TITLE
Flashback: add recipe

### PR DESCRIPTION
**Pull request by Luca1991**
This is a recipe for Flashback.
This is based on the opensource reimplementation called  REminiscence REinforced.
Launch script + icon included.

You need original game files to play (I tested the DOS version).

@Begasus  I didn't forget the issue with 32bit build of melonDS, it is just that I have a couple of recipes already  ready.

### CODE REVIEW

File: games-action/flashback/flashback-0.5.2.recipe [hunk 1]
```diff
@@ -0,0 +1,61 @@
+SUMMARY="Based on REminiscence REinforced"
+DESCRIPTION="REminiscence is a re-implementation of the engine used in
+the game 'Flashback: The Quest For Identity' made by Delphine Software and released in 1992.
+REinforced is an enhanced version of the original REminiscence.
+
+You will need the original files of the PC (DOS or CD), Amiga or Macintosh release."
+HOMEPAGE="http://cyxdown.free.fr/reminiscence"
+COPYRIGHT="2005-2021 Gregory Montoir
+2020-2021 Alex Chermenin"
```
Comment (Begasus) [inline games-action/flashback/flashback-0.5.2.recipe:9]: 1 tab in front of the line

File: games-action/flashback/flashback-0.5.2.recipe [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="https://github.com/chermenin/REminiscence/archive/refs/tags/$portVersion.tar.gz"
+CHECKSUM_SHA256="c3b5a0d074e6846499e641fee27384b0a09769be6634215abd82d4d78691f8e8"
+SOURCE_DIR="REminiscence-$portVersion"
+ADDITIONAL_FILES="flashback.hvif
+flashback.sh"
```
Comment (Begasus) [inline games-action/flashback/flashback-0.5.2.recipe:16]: 1 tab in front of this line too.

**Comment by Luca1991** [comment]
Proposed changes implemented ;)

File: games-action/flashback/flashback-0.5.2.recipe [hunk 3]
```diff
@@ -0,0 +1,61 @@
+SUMMARY="Based on REminiscence REinforced"
+DESCRIPTION="REminiscence is a re-implementation of the engine used in
+the game 'Flashback: The Quest For Identity' made by Delphine Software and released in 1992.
```
Comment (Begasus) [inline games-action/flashback/flashback-0.5.2.recipe:3]: Wrap lines up to 100 characters, cut line at the end with a backslash.

File: games-action/flashback/flashback-0.5.2.recipe [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	flashback$secondaryArchSuffix = $portVersion
+	app:Flashback = $portVersion
+	cmd:flashback
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	tremor$secondaryArchSuffix
```
Comment (Begasus) [inline games-action/flashback/flashback-0.5.2.recipe:28]: There is a library there :)
```diff
-	tremor$secondaryArchSuffix
```

Comment (Begasus) [reply games-action/flashback/flashback-0.5.2.recipe:28]: Use 1 tab in front of the lines (several spaces in the recipe atm).

File: games-action/flashback/flashback-0.5.2.recipe [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+	lib:libmodplug$secondaryArchSuffix
+        lib:libogg$secondaryArchSuffix
+        lib:libSDL2_2.0$secondaryArchSuffix
```
Comment (Begasus) [inline games-action/flashback/flashback-0.5.2.recipe:31]: Use tremors library:
```diff
+	lib:libvorbisidec$secondaryArchSuffix
```

File: games-action/flashback/flashback-0.5.2.recipe [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+        lib:libz$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	tremor${secondaryArchSuffix}_devel
```
Comment (Begasus) [inline games-action/flashback/flashback-0.5.2.recipe:37]: Same here:
```diff
-	tremor${secondaryArchSuffix}_devel
```

File: games-action/flashback/flashback-0.5.2.recipe [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+	devel:libmodplug$secondaryArchSuffix
+	devel:libSDL2_2.0$secondaryArchSuffix
```
Comment (Begasus) [inline games-action/flashback/flashback-0.5.2.recipe:39]: Here too:
```diff
+	devel:libvorbisidec$secondaryArchSuffix
```

File: games-action/flashback/flashback-0.5.2.recipe [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+BUILD()
+{
+	make $jobArgs
+}
+
+INSTALL()
+{
+	mkdir -p $binDir $appsDir
+	cp fb "$appsDir/Flashback"
+	cp $portDir/additional-files/flashback.sh "$binDir/flashback"
```
Comment (Begasus) [inline games-action/flashback/flashback-0.5.2.recipe:56]: Better copy this to `$prefix/bin`, doing this in `$bindir` ends up in `$prefix/bin/x86` on 32bit secondary architecture and then needs `cmd:flashback$secondaryArchSuffix` in PROVIDES then.

**Review by Begasus — CHANGES_REQUESTED** [review]

**Comment by Luca1991** [comment]
Proposed changes implemented

**Comment by Luca1991** [comment]
@Begasus is there anything else preventing the merge? Thanks

**Comment by Begasus** [comment]
> @Begasus is there anything else preventing the merge? Thanks

Maybe also add the IOM file in here too?

**Comment by Luca1991** [comment]
@Begasus it is an SVG converted to HVIF. Is it possible to use an rdef to apply an icon to an sh script?

**Comment by Begasus** [comment]
> @Begasus it is an SVG converted to HVIF. Is it possible to use an rdef to apply an icon to an sh script?

Maybe better export to rdef then, yes you can use rdef to apply the icon to a script, see:
https://github.com/Begasus/haikuports/wiki/Useful-variables-for-recipes#rename-rdef-to-rsrc-usefull-sometimes-for-scripts-that-cant-use-rdef-files

**Comment by Luca1991** [comment]
@Begasus now it should be ok.

**Comment by Begasus** [comment]
Thanks, keep an eye out on the buildmasters for build progress :)

**Pull request merged**